version: '3.9'

services:
  flask-app:
    command: python3 app.py  
    image: amanbh858/my_model:latest 
    platform: linux/x86_64  
    environment:
      HOST: 0.0.0.0  
      PORT: 5000  
    # Traefik routing and labels
    labels:
      traefik.enable: "true"  # Enable Traefik for routing
      traefik.http.services.flask_app-service.loadbalancer.server.port: "5000" 
      traefik.http.routers.flask_app-router.rule: Host(`yourdomain.com`)  
      traefik.http.routers.flask_app-router.tls: "true"  
      traefik.http.routers.flask_app-router.tls.certresolver: "production"  
      traefik.http.routers.flask_app-router.entrypoints: "websecure" 
    networks:
      - default
      - traefik_default  # Ensure this network exists for Traefik routing

# Network configuration
networks:
  default:
  traefik_default:
    external: true  # Connect to the external Traefik network for routing
